### ISSUE TITLE
[Expo-Notifications] Permissions.NOTIFICATIONS return undetermined on iOS.

### ISSUE THREAD

**Issue by anhquan291** [expo/expo#10374]
## 🐛 Bug Report
Hi Guys,
I'm having an issue with the Permissions.NOTIFICATIONS on iOS (but works well on Android). It was working fine for a few months. However, now It returns undetermined on iOS devices (both Expo cli or standalone app). 

```bash
const AskingNotificationPermissonToken = async () => {
  const { status: existingStatus } = await Permissions.getAsync(
    Permissions.NOTIFICATIONS
  );
  let finalStatus = existingStatus;

  if (existingStatus !== "granted") {
    const { status } = await Permissions.askAsync(Permissions.NOTIFICATIONS);
    finalStatus = status;
    console.log(finalStatus);
  }
  if (finalStatus !== "granted") {
    console.log("Failed to get push token for push notification!");
    return (token = "");
  }
  token = (await Notifications.getExpoPushTokenAsync()).data;
  return token;
};
```

### Environment - output of `expo diagnostics` & the platform(s) you're targeting

Expo CLI 3.27.8 environment info:
System:
OS: macOS Catalina
Shell: 3.2.57 - /bin/bash
Binaries:
Node: 10.22.0 - /usr/local/bin/node
Yarn: 1.22.4 - /usr/local/bin/yarn
npm: 6.14.6 - /usr/local/bin/npm
Watchman: 4.9.0 - /usr/local/bin/watchman
Managers:
CocoaPods: 1.8.4 - /usr/local/bin/pod
SDKs:
iOS SDK:
Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
Android SDK:
API Levels: 23, 28, 29
Build Tools: 28.0.3, 29.0.2, 29.0.3
System Images: android-28 | Intel x86 Atom, android-29 | Google APIs Intel x86 Atom
IDEs:
Android Studio: 4.0 AI-193.6911.18.40.6514223
Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
npmPackages:
expo: ~39.0.2 => 39.0.3
react: 16.13.1 => 16.13.1
react-dom: 16.13.1 => 16.13.1
react-native: ~0.63.2 => 0.63.2
react-native-web: ~0.13.12 => 0.13.13
npmGlobalPackages:
expo-cli: 3.27.8
Expo Workflow: managed

### Expected Behavior vs Actual Behavior
Expected Behavior: return "granted"
Actual Behavior: return "undetermined"

Thanks in Advanced

**Comment by cruzach** [comment]
What versions of `expo-notifications` and `expo-permissions` do you have installed? please share a reproducible demo

**Comment by anhquan291** [comment]
Hi, @cruzach ,
I'm using:
import * as Notifications from "expo-notifications";
import * as Permissions from "expo-permissions";
```bash
   "expo-notifications": "~0.3.3",
    "expo-permissions": "~9.0.1",
```
Everything worked perfect til last week sir(still works well on android). It's so strange. Thanks in advance

**Comment by cruzach** [comment]
So it was working until the upgrade? again, we need a reproducible demo to investigate this

**Comment by anhquan291** [comment]
> So it was working until the upgrade? 

No sir, I'm still in SDK 38. I'll try to reinstall those packages.

**Comment by anhquan291** [comment]
Updating: 
I've tried with the other Permissions and they work just fine. 
```bash
The await Permissions.askAsync(Permissions.NOTIFICATIONS);
```
The Permissions.askAsync for notification just doesn't ask for anything evenwhen the status is "undetermined"

**Comment by cruzach** [comment]
okay sorry my misunderstanding, your environment shows you're on SDK 39:
> expo: ~39.0.2 => 39.0.3

It's not totally clear what happened here. From above, it seems like without any changes to your standalone app, the permission all of a sudden changed to returning `undetermined`? You made no updates to the code at all?

**Comment by anhquan291** [comment]
@cruzach no sir, I haven't touch to the AskPermission component at all. It just happens. It still works well on android. So weird. 
Have a good day sir.

**Comment by anhquan291** [comment]
Hi @cruzach,
Sorry for bothering you again. You know I've created a new project today for testing the permissions notification that I'm fighting with, and the same issue happens with iOS, no prompt dialog shows up, status always returns undetermined.
Hope to hear good news from u soon. Thank you

**Comment by cruzach** [comment]
Can you try uninstalling and reinstalling the Expo client app, and run [this example](https://snack.expo.io/@charliecruzan/push-notifications) and let me know if it works?

**Comment by anhquan291** [comment]
> Can you try uninstalling and reinstalling the Expo client app, and run [this example](https://snack.expo.io/@charliecruzan/push-notifications) and let me know if it works?

Thank you so much for being so patient with my issues. 
I opened your examples on simulator as well as physic devices, unfortunately it failed. :-( it's so strange. Have a great sir. 
![image](https://user-images.githubusercontent.com/17813202/94701930-64a54680-0367-11eb-9116-a5405048d881.png)

**Comment by cruzach** [comment]
did you uninstall and reinstall the Expo client app?

**Comment by anhquan291** [comment]
> did you uninstall and reinstall the Expo client app?

I did it before but no luck, but now I just opened it again on my phone and it works!!! So I'm going to build the standalone for the iOS simulator now. Let's hope it works. 
Thank you so much sir. Really appreciate that.

**Comment by cruzach** [comment]
local notifications will work on android emulators, but not remote notifications

I'm going to close this issue seen it seems it was just an intermittent problem

**Comment by nathanwaters** [comment]
Having the same problem. Here's a workaround for now that behaves as expected in Expo client app on iOS:

`await Permissions.getAsync(Permissions.NOTIFICATIONS)` always returns "undetermined".
`await Notifications.getPermissionsAsync()` returns correctly.

**Comment by cruzach** [comment]
@nathanwaters I'm happy to if you can provide a reproducible demo via [snack](https://snack.expo.io/)

**Comment by nathanwaters** [comment]
@cruzach no worries, give it a spin: https://snack.expo.io/xUTKs6vSb

**Comment by cruzach** [comment]
On a clean install of the Expo client app on an iPhone 6, on first open, both return `undetermined` (which is expected, since we haven't asked for permission yet). After adding  `Permissions.askAsync(Permissions.NOTIFICATIONS);` to line 28, I see the prompt, say yes, and then both resolve to `granted` (which is expected).

**Comment by nathanwaters** [comment]
Hmm ok so that flow works. I added an "ask permissions" button: https://snack.expo.io/@nathanwaters/expo-notifications-vs-expo-permissions-undetermined-bug

Try this flow:

1. Fresh Expo client install
2. Tap "ask permissions"
3. Don't allow
4. Tap "open settings"
5. Enable notifications
6. Return to app

**Comment by cruzach** [comment]
yeah I reproduced it using that flow- for now you can simply kill the app and re-open it, and then the `expo-permissions` permission **also** resolved to `granted` for me. Lmk if you see the same thing @nathanwaters

**Comment by cruzach** [comment]
I think a new issue report is necessary since we've narrowed this down to iOS only

**Comment by nathanwaters** [comment]
Still shows "undetermined" for me after killing and re-opening. Yeah good idea @cruzach here's a new issue: https://github.com/expo/expo/issues/10735
